ibm/service/cos/resource_ibm_cos_bucket_lifecycle_configuration.go

```diff
@@ -0,0 +1,568 @@
+	bucketCRN := parseWebsiteId(d.Id(), "bucketCRN")
+	bucketName := parseWebsiteId(d.Id(), "bucketName")
+	bucketLocation := parseWebsiteId(d.Id(), "bucketLocation")
+	instanceCRN := parseWebsiteId(d.Id(), "instanceCRN")
+	endpointType := parseWebsiteId(d.Id(), "endpointType")
+	d.Set("bucket_crn", bucketCRN)
+	d.Set("bucket_location", bucketLocation)
+	if endpointType != "" {
+		d.Set("endpoint_type", endpointType)
+	}
+	bxSession, err := meta.(conns.ClientSession).BluemixSession()
+	if err != nil {
+		return err
+	}
+	s3Client, err := getS3ClientSession(bxSession, bucketLocation, endpointType, instanceCRN)
+	if err != nil {
+		return err
+	}
+	//getBucketConfiguration
+	const (
+		lifecycleConfigurationRulesSteadyTimeout = 2 * time.Minute
+	)
+	getLifecycleConfigurationInput := &s3.GetBucketLifecycleConfigurationInput{
+		Bucket: aws.String(bucketName),
+	}
+	var output *s3.GetBucketLifecycleConfigurationOutput
+
+	// Adding a retry to overcome the
+	err = resource.Retry(lifecycleConfigurationRulesSteadyTimeout, func() *resource.RetryError {
+		var err error
+		output, err = s3Client.GetBucketLifecycleConfiguration(getLifecycleConfigurationInput)
+
+		if d.IsNewResource() && err != nil && strings.Contains(err.Error(), "NoSuchLifecycleConfiguration: The lifecycle configuration does not exist") {
+
+			return resource.RetryableError(err)
+
+		}
+
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+	if conns.IsResourceTimeoutError(err) {
+		output, err = s3Client.GetBucketLifecycleConfiguration(getLifecycleConfigurationInput)
+	}
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error getting Lifecycle Configuration for the bucket %s", bucketName)
+	}
+
+	var outputptr *s3.LifecycleConfiguration
+	outputptr = (*s3.LifecycleConfiguration)(output)
+	if err != nil && !strings.Contains(err.Error(), "AccessDenied: Access Denied") {
+		return err
+	}
+	if outputptr.Rules != nil {
+		lifecycleConfiguration := flex.LifecylceRuleGet(outputptr.Rules)
+		if len(lifecycleConfiguration) > 0 {
+			d.Set("lifecycle_rule", lifecycleConfiguration)
+		}
+	} else {
+		d.SetId("")
+	}
+	return nil
+}
+
+func resourceIBMCOSBucketLifecycleConfigurationDelete(d *schema.ResourceData, meta interface{}) error {
+	bucketName := parseWebsiteId(d.Id(), "bucketName")
+	bucketLocation := parseWebsiteId(d.Id(), "bucketLocation")
+	instanceCRN := parseWebsiteId(d.Id(), "instanceCRN")
+	endpointType := parseWebsiteId(d.Id(), "endpointType")
+	bxSession, err := meta.(conns.ClientSession).BluemixSession()
+	if err != nil {
+		return err
+	}
+	s3Client, err := getS3ClientSession(bxSession, bucketLocation, endpointType, instanceCRN)
+	if err != nil {
+		return err
+	}
+	deleteBucketLifecycleInput := &s3.DeleteBucketLifecycleInput{
+		Bucket: aws.String(bucketName),
+	}
+	_, err = s3Client.DeleteBucketLifecycle(deleteBucketLifecycleInput)
+	if err != nil {
```

Comment:
shall we introduce a timeout check for bucket deletion as well , like we have done in the case of read. Just to ensure proper deletion in case the config is not found immediately after creation ? 
CC @avinash1IBM

Reply:
A timeout in delete is not required , as I haven't encountered any issue with the delete functionality. And delete is called  once the resource are created and exist.And in read we are already making sure we get a lifecycle configuration.

Reply:
Yes, I agree with @IBM-diksha, we no need to wait till the policy is deleted instead a 200 is fine.